tests/sqlparser_hive.rs

```diff
@@ -537,6 +537,33 @@ fn parse_use() {
     );
 }
 
+#[test]
+fn parse_hive_unary_ops() {
```

Comment:
one general comment regarding the tests, since we're relying on the dialect api to guard the functionality, can we move the tests to common.rs and write them in this style?
https://github.com/apache/datafusion-sqlparser-rs/blob/main/tests/sqlparser_common.rs#L2781

Also we should do the same for the factorial tests since the current PR seems to be introducing two APIs to the dialect trait

Reply:
Good idea! I will try it. :)

Reply:
The test comments are also unadressed?